stock-tracker/src/features/chart/component/chartButton.tsx

```diff
@@ -1,9 +1,10 @@
 import React, { FC } from "react";
+import { ChartTimes } from "../redux/actions";
 
 type ChartButtonProps = {
   current: string;
-  range: string;
-  onClick: React.MouseEventHandler<HTMLButtonElement>;
+  range: ChartTimes;
+  onClick: <K extends ChartTimes>(chartTime: K) => void;
```

Comment:
This generic doesn't really give you any benefit here, without linking it to the `range` prop:
```ts
type ChartButtonProps<CT extends ChartTimes> = {
  current: ChartTimes;
  range: CT;
  onClick: (chartTime: CT) => void;
}
```